crates/app/src/lib.rs

```diff
@@ -27,6 +29,8 @@ pub const APP_DESCRIPTION_KEY: MetadataKey = MetadataKey::new("description");
 /// MetadataKey for extracting the OCI image digest.
 pub const OCI_IMAGE_DIGEST_KEY: MetadataKey = MetadataKey::new("oci_image_digest");
 
+type ValidatorFn = dyn Fn(&App, &[String]) -> anyhow::Result<()>;
```

Comment:
As this is used in a public interface, I wonder if it needs a `pub` or a doc comment or something?  I'm not sure what the expectation is with type aliases (and clearly the compiler doesn't mind).